pkgs/by-name/di/dirstack/package.nix

```diff
@@ -0,0 +1,44 @@
+{
+  fetchFromGitHub,
+  fzf,
+  lib,
+  rustPlatform,
+  stdenv,
+}:
+let
+  src = fetchFromGitHub {
+    owner = "NewDawn0";
+    repo = "dirStack";
+    rev = "8ef0f19ae366868fb046f6acb0d396bc8436ef31";
+    hash = "sha256-mzw3uDZ0eM81WJM0YNcOlMHs4fNMUBgwxS6VBSS+VS8=";
+  };
+  meta = with lib; {
+    description = "A cd quicklist";
+    homepage = "https://github.com/NewDawn0/dirStack";
+    maintainers = with maintainers; [ NewDawn0 ];
+    license = licenses.mit;
+  };
+  pkg = rustPlatform.buildRustPackage {
+    inherit meta src;
+    pname = "dirStack";
+    version = "1.0.0";
+    propagatedBuildInputs = [ fzf ];
+    cargoLock.lockFile = "${src}/Cargo.lock";
```

Comment:
Why is this necessary? (Set cargoHash instead)

Reply:
I figured If I updated the package, I would have to update less since it would use the new `Cargo.Lock` from the `${src}` and would thereby only have to update the `Cargo.lock`. Please let me know I should change it to use `cargoHash` instead of `lockFile`

Reply:
lockFile is specifically used for lock files we have vendored inside Nixpkgs for whatever reason (upstream's lockfile is missing/wrong/outdated/etc) — if you're using a lock file from the package itself, just use cargoHash.

(Also, I think using lockFile like this will introduce an import-from-derivation (IFD), which is not permitted in Nixpkgs, so there's really no reason to do this)